ext/intl/listformatter/listformatter_class.c

```diff
@@ -0,0 +1,241 @@
+        zend_throw_exception(IntlException_ce_ptr, "Constructor failed", 0);
+        RETURN_THROWS();
+    }
+}
+
+PHP_METHOD(IntlListFormatter, format)
+{
+    ListFormatter_object *obj = Z_INTL_LISTFORMATTER_P(ZEND_THIS);
+    HashTable *ht;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_ARRAY_HT(ht)
+    ZEND_PARSE_PARAMETERS_END();
+
+    if (!LISTFORMATTER_OBJECT(obj)) {
+        zend_throw_exception(NULL, "ListFormatter not properly constructed", 0);
+        RETURN_FALSE;
+    }
+
+    uint32_t count = zend_hash_num_elements(ht);
+    if (count == 0) {
+        RETURN_EMPTY_STRING();
+    }
+
+    const UChar **items = (const UChar **)safe_emalloc(count, sizeof(const UChar *), 0);
+    int32_t *itemLengths = (int32_t *)safe_emalloc(count, sizeof(int32_t), 0);
+    uint32_t i = 0;
+    zval *val;
+
+    ZEND_HASH_FOREACH_VAL(ht, val) {
+        zend_string *str_val;
+        
+        str_val = zval_get_string(val);
+        
+        // Convert PHP string to UTF-16
+        UChar *ustr = NULL;
+        int32_t ustr_len = 0;
+        UErrorCode status = U_ZERO_ERROR;
+        
+        intl_convert_utf8_to_utf16(&ustr, &ustr_len, ZSTR_VAL(str_val), ZSTR_LEN(str_val), &status);
+        zend_string_release(str_val);
+
+        if (U_FAILURE(status)) {
+            // We can't use goto cleanup because items and itemLengths are incompletely allocated
+            for (uint32_t j = 0; j < i; j++) {
+                efree((void *)items[j]);
+            }
+            efree(items);
+            efree(itemLengths);
+            intl_error_set(NULL, status, "Failed to convert string to UTF-16", 0);
+            RETURN_FALSE;
+        }
+        
+        items[i] = ustr;
+        itemLengths[i] = ustr_len;
+        i++;
+    } ZEND_HASH_FOREACH_END();
+
+    UErrorCode status = U_ZERO_ERROR;
+    int32_t resultLength;
+    UChar *result = NULL;
+
+    resultLength = ulistfmt_format(LISTFORMATTER_OBJECT(obj), items, itemLengths, count, NULL, 0, &status);
+
+    if (U_FAILURE(status) && status != U_BUFFER_OVERFLOW_ERROR) {
+        intl_error_set(NULL, status, "Failed to format list", 0);
+        RETVAL_FALSE;
+        goto cleanup;
+    }
+
+    // Allocate buffer and try again
+    status = U_ZERO_ERROR;
+    result = (UChar *)safe_emalloc(resultLength + 1, sizeof(UChar), 0);
+    ulistfmt_format(LISTFORMATTER_OBJECT(obj), items, itemLengths, count, result, resultLength, &status);
+
+    if (U_FAILURE(status)) {
+        if (result) {
+            efree(result);
+        }
+        intl_error_set(NULL, status, "Failed to format list", 0);
+        RETVAL_FALSE;
+        goto cleanup;
+    }
+
+    // Convert result back to UTF-8
+    zend_string *ret = intl_convert_utf16_to_utf8(result, resultLength, &status);
+    efree(result);
+    
+    if (!ret) {
+        intl_error_set(NULL, status, "Failed to convert result to UTF-8", 0);
+        RETVAL_FALSE;
+    } else {
+        RETVAL_NEW_STR(ret);
+    }
+
+    cleanup:
```

Comment:
label ought to be on the far left. below code then can be aligned properly with the rest